atom/renderer/api/lib/desktop-capturer.coffee

```diff
@@ -0,0 +1,10 @@
+ipc = require 'ipc'
+NativeImage = require 'native-image'
+
+getSources = (options, callback) ->
+  ipc.send 'ATOM_BROWSER_DESKTOP_CAPTURER_GET_SOURCES', options
+  ipc.once 'ATOM_RENDERER_DESKTOP_CAPTURER_RESULT', (sources) ->
```

Comment:
> @zcbenz, looks like the behavior EventEmitter.once in electron is different than it in node. In node, this listener-limitted warning will not appear if I call once more than 10 times while it appears in electron.

Node will show the warning, you can try:

```js
for (var i = 0; i < 20; ++i) process.once('test', function(){});
```

Also current implement have problems if you call `getSources` for multiple times at once:

```coffeescript
dc.getSources {}, -> console.log('done')
dc.getSources {}, -> console.log('done')
```

Each callback will be called twice since `ATOM_RENDERER_DESKTOP_CAPTURER_RESULT` is sent back twice.

We can probably assign each `getSources` call an ID, and only listen to the response with the ID:

```coffeescript
id = getNextID()
ipc.send 'ATOM_BROWSER_DESKTOP_CAPTURER_GET_SOURCES', options, id
ipc.once "ATOM_RENDERER_DESKTOP_CAPTURER_RESULT_#{id}", (sources) ->
```